invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/state/actions/deposit.js

```diff
@@ -63,10 +63,34 @@ export const saveDraftWithUrlUpdate = async (draft, draftsService) => {
   return response;
 };
 
+function _hasValidationErrorsWithSeverityError(errors) {
+  if (typeof errors === "object") {
+    if (Object.hasOwn(errors, "message") && Object.hasOwn(errors, "severity")) {
+      if (errors["severity"] === "error") {
+        return true;
+      }
+    }
+    for (const key of Object.keys(errors)) {
+      if (key !== "message" && key !== "severity") {
+        return _hasValidationErrorsWithSeverityError(errors[key]);
+      }
+    }
+  } else {
+    // If the error message is a string and not an object with `message` and `severity` keys, then it's an error.
+    return true;
+  }
+}
+
```

Comment:
for readability:
```suggestion
function _hasValidationErrorsWithSeverityError(errors) {
  const isObjectError = typeof errors === "object";
  const isSeverityError = errors["severity"] === "error";
  
  if (isObjectError && isSeverityError) {
        return true;
    }
    for (const key of Object.keys(errors)) {
    \\ we need an explanation here on why we do this, I am not understanding why :)
      if (key !== "message" && key !== "severity") {
        return _hasValidationErrorsWithSeverityError(errors[key]);
      }
    }
    
   // If the error message is a string and not an object with `message` and `severity` keys, then it's an error.
   return true;
}

```

i removed:
```
if (Object.hasOwn(errors, "message") && Object.hasOwn(errors, "severity")) 
```
because this can obfuscate the fact that we are getting some other object than we expect as the error. In this case we should fail, not hide it :)
also, it helps with readability